zmq/tests/cython_ext.pyx missing from install

I'm packaging your module as an rpm package so I'm using the typical PEP517 based build, install and test cycle used on building packages from non-root account.
- `python3 -sBm build -w --no-isolation`
- because I'm calling `build` with `--no-isolation` I'm using during all processes only locally installed modules
- install .whl file in </install/prefix>
- run pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>

pytest is failing in two units. There are as well some warnings.
Here is pytest output:
<details>

```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra --import-mode=importlib
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.14, pytest-7.1.3, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0, configfile: pytest.ini, testpaths: zmq/tests
collected 357 items / 1 skipped

zmq/tests/test_asyncio.py ....................................                                                                                                       [ 10%]
zmq/tests/test_auth.py .................                                                                                                                             [ 14%]
zmq/tests/test_cffi_backend.py ssssssssssssss                                                                                                                        [ 18%]
zmq/tests/test_constants.py ..........................................                                                                                               [ 30%]
zmq/tests/test_context.py ....................s......                                                                                                                [ 38%]
zmq/tests/test_cython.py FF                                                                                                                                          [ 38%]
zmq/tests/test_decorators.py ...........................                                                                                                             [ 46%]
zmq/tests/test_device.py ........                                                                                                                                    [ 48%]
zmq/tests/test_draft.py ss                                                                                                                                           [ 49%]
zmq/tests/test_error.py ...                                                                                                                                          [ 49%]
zmq/tests/test_etc.py .ss                                                                                                                                            [ 50%]
zmq/tests/test_ext.py .                                                                                                                                              [ 50%]
zmq/tests/test_future.py ....................                                                                                                                        [ 56%]
zmq/tests/test_imports.py ..........s.s.s.s.s.s.s                                                                                                                    [ 63%]
zmq/tests/test_includes.py ..                                                                                                                                        [ 63%]
zmq/tests/test_ioloop.py ....                                                                                                                                        [ 64%]
zmq/tests/test_log.py ........                                                                                                                                       [ 66%]
zmq/tests/test_message.py ....................                                                                                                                       [ 72%]
zmq/tests/test_monitor.py ......                                                                                                                                     [ 74%]
zmq/tests/test_monqueue.py ........                                                                                                                                  [ 76%]
zmq/tests/test_multipart.py ..                                                                                                                                       [ 77%]
zmq/tests/test_pair.py ....                                                                                                                                          [ 78%]
zmq/tests/test_poll.py ........                                                                                                                                      [ 80%]
zmq/tests/test_proxy_steerable.py ...                                                                                                                                [ 81%]
zmq/tests/test_pubsub.py ..                                                                                                                                          [ 81%]
zmq/tests/test_reqrep.py ......                                                                                                                                      [ 83%]
zmq/tests/test_retry_eintr.py s...s.                                                                                                                                 [ 85%]
zmq/tests/test_security.py .....                                                                                                                                     [ 86%]
zmq/tests/test_socket.py .............................s.....                                                                                                         [ 96%]
zmq/tests/test_ssh.py .                                                                                                                                              [ 96%]
zmq/tests/test_version.py ....                                                                                                                                       [ 97%]
zmq/tests/test_win32_shim.py s                                                                                                                                       [ 98%]
zmq/tests/test_z85.py ....                                                                                                                                           [ 99%]
zmq/tests/test_zmqstream.py ...                                                                                                                                      [100%]

================================================================================= FAILURES =================================================================================
______________________________________________________________________________ test_cython[3] ______________________________________________________________________________

language_level = 3, request = <FixtureRequest for <Function test_cython[3]>>, tmpdir = local('/tmp/pytest-of-tkloczko/pytest-997/test_cython_3_0')

    @pytest.mark.skipif(
        'zmq.backend.cython' not in sys.modules, reason="Requires cython backend"
    )
    @pytest.mark.skipif(
        sys.platform.startswith('win'), reason="Don't try runtime Cython on Windows"
    )
    @pytest.mark.parametrize('language_level', [3, 2])
    def test_cython(language_level, request, tmpdir):
        import pyximport

        assert 'zmq.tests.cython_ext' not in sys.modules

        importers = pyximport.install(
            setup_args=dict(include_dirs=zmq.get_includes()),
            language_level=language_level,
            build_dir=str(tmpdir),
        )

        cython_ext = None

        def unimport():
            pyximport.uninstall(*importers)
            sys.modules.pop('zmq.tests.cython_ext', None)

        request.addfinalizer(unimport)

        # this import tests the compilation
>       from . import cython_ext
E       ImportError: cannot import name 'cython_ext' from 'zmq.tests' (/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/zmq/tests/__init__.py)

zmq/tests/test_cython.py:35: ImportError
______________________________________________________________________________ test_cython[2] ______________________________________________________________________________

language_level = 2, request = <FixtureRequest for <Function test_cython[2]>>, tmpdir = local('/tmp/pytest-of-tkloczko/pytest-997/test_cython_2_0')

    @pytest.mark.skipif(
        'zmq.backend.cython' not in sys.modules, reason="Requires cython backend"
    )
    @pytest.mark.skipif(
        sys.platform.startswith('win'), reason="Don't try runtime Cython on Windows"
    )
    @pytest.mark.parametrize('language_level', [3, 2])
    def test_cython(language_level, request, tmpdir):
        import pyximport

        assert 'zmq.tests.cython_ext' not in sys.modules

        importers = pyximport.install(
            setup_args=dict(include_dirs=zmq.get_includes()),
            language_level=language_level,
            build_dir=str(tmpdir),
        )

        cython_ext = None

        def unimport():
            pyximport.uninstall(*importers)
            sys.modules.pop('zmq.tests.cython_ext', None)

        request.addfinalizer(unimport)

        # this import tests the compilation
>       from . import cython_ext
E       ImportError: cannot import name 'cython_ext' from 'zmq.tests' (/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/zmq/tests/__init__.py)

zmq/tests/test_cython.py:35: ImportError
============================================================================= warnings summary =============================================================================
zmq/tests/test_poll.py:160
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_poll.py:160: PytestUnknownMarkWarning: Unknown pytest.mark.flaky - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @mark.flaky(reruns=3)

zmq/tests/test_poll.py:195
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_poll.py:195: PytestUnknownMarkWarning: Unknown pytest.mark.flaky - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @mark.flaky(reruns=3)

zmq/tests/test_retry_eintr.py:60
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_retry_eintr.py:60: PytestUnknownMarkWarning: Unknown pytest.mark.flaky - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @mark.flaky(reruns=3)

zmq/tests/test_auth.py: 10 warnings
zmq/tests/test_future.py: 20 warnings
zmq/tests/test_ioloop.py: 1 warning
zmq/tests/test_zmqstream.py: 3 warnings
  /usr/lib64/python3.8/site-packages/tornado/ioloop.py:350: DeprecationWarning: make_current is deprecated; start the event loop first
    self.make_current()

zmq/tests/test_context.py::TestContext::test_fail_init
  /usr/lib/python3.8/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function Context.__del__ at 0x7f4e7cbd29d0>

  Traceback (most recent call last):
    File "/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/zmq/sugar/context.py", line 93, in __del__
      self.destroy()
    File "/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/zmq/sugar/context.py", line 258, in destroy
      sockets = self._sockets
    File "/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/zmq/sugar/attrsettr.py", line 48, in __getattr__
      raise AttributeError(
  AttributeError: Context has no such option: _SOCKETS

    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))

zmq/tests/test_cython.py::test_cython[3]
  /usr/lib64/python3.8/site-packages/pyximport/pyximport.py:51: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

zmq/tests/test_future.py: 20 warnings
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_future.py:25: DeprecationWarning: make_current is deprecated; start the event loop first
    self.loop.make_current()

zmq/tests/test_future.py: 20 warnings
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_future.py:32: DeprecationWarning: clear_current is deprecated
    IOLoop.clear_current()

zmq/tests/test_future.py: 20 warnings
zmq/tests/test_ioloop.py: 4 warnings
  /usr/lib64/python3.8/site-packages/tornado/ioloop.py:227: DeprecationWarning: clear_current is deprecated
    IOLoop.clear_current()

zmq/tests/test_ioloop.py::TestIOLoop::test_close_all
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_ioloop.py:111: DeprecationWarning: zmq.eventloop.ioloop is deprecated in pyzmq 17. pyzmq now works with default tornado and asyncio eventloops.
    loop = self.IOLoop.current()

zmq/tests/test_ioloop.py::TestIOLoop::test_close_all
zmq/tests/test_ioloop.py::TestIOLoop::test_current
zmq/tests/test_ioloop.py::TestIOLoop::test_instance
zmq/tests/test_ioloop.py::TestIOLoop::test_simple
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_ioloop.py:74: DeprecationWarning: clear_current is deprecated
    BaseIOLoop.clear_current()

zmq/tests/test_ioloop.py::TestIOLoop::test_simple
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_ioloop.py:80: DeprecationWarning: make_current is deprecated; start the event loop first
    loop.make_current()

zmq/tests/test_monitor.py::TestSocketMonitorAsyncIO::test_monitor
zmq/tests/test_monitor.py::TestSocketMonitorAsyncIO::test_monitor_connected
  /home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/zmq/_future.py:663: RuntimeWarning: No running event loop. zmq.asyncio should be used from within an asyncio loop.
    self._update_handler(self._state)

zmq/tests/test_socket.py::TestSocket::test_int_sockopts
zmq/tests/test_socket.py::TestSocket::test_int_sockopts
zmq/tests/test_socket.py::TestSocket::test_mockable
  /home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/zmq/sugar/socket.py:261: DeprecationWarning: Socket.socket_type is deprecated, use Socket.type
    warn("Socket.socket_type is deprecated, use Socket.type", DeprecationWarning)

zmq/tests/test_zmqstream.py::TestZMQStream::test_callable_check
zmq/tests/test_zmqstream.py::TestZMQStream::test_on_recv_basic
zmq/tests/test_zmqstream.py::TestZMQStream::test_on_recv_wake
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_zmqstream.py:29: DeprecationWarning: make_current is deprecated; start the event loop first
    self.loop.make_current()

zmq/tests/test_zmqstream.py::TestZMQStream::test_callable_check
zmq/tests/test_zmqstream.py::TestZMQStream::test_on_recv_basic
zmq/tests/test_zmqstream.py::TestZMQStream::test_on_recv_wake
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_zmqstream.py:39: DeprecationWarning: clear_current is deprecated
    ioloop.IOLoop.clear_current()

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================= short test summary info ==========================================================================
SKIPPED [1] zmq/tests/test_mypy.py:12: could not import 'mypy': No module named 'mypy'
SKIPPED [1] zmq/tests/test_cffi_backend.py:83: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:93: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:34: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:62: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:133: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:108: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:122: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:115: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:210: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:174: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:146: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:49: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:40: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:29: CFFI not available
SKIPPED [1] zmq/tests/test_context.py:351: Requires pyczmq
SKIPPED [2] zmq/tests/test_draft.py:15: draft api unavailable
SKIPPED [1] zmq/tests/test_etc.py:17: bundled libzmq
SKIPPED [1] zmq/tests/test_etc.py:23: bundled libzmq
SKIPPED [7] zmq/tests/test_imports.py:93: could not import 'zmq.green': No module named 'gevent'
SKIPPED [1] zmq/tests/test_retry_eintr.py:90: TODO: find a way to interrupt getsockopt
SKIPPED [1] zmq/tests/test_retry_eintr.py:93: TODO: find a way to interrupt setsockopt
SKIPPED [1] zmq/tests/test_socket.py:539: Requires pyczmq
SKIPPED [1] zmq/tests/test_win32_shim.py:25: Windows only test
FAILED zmq/tests/test_cython.py::test_cython[3] - ImportError: cannot import name 'cython_ext' from 'zmq.tests' (/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc...
FAILED zmq/tests/test_cython.py::test_cython[2] - ImportError: cannot import name 'cython_ext' from 'zmq.tests' (/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc...
==================================================== 2 failed, 325 passed, 31 skipped, 120 warnings in 67.87s (0:01:07) ====================================================
```
</details>

Here is list of installed modules in build env
<details>

```console
Package                       Version
----------------------------- -----------------
alabaster                     0.7.12
apeye                         1.2.0
appdirs                       1.4.4
attrs                         22.1.0
autodocsumm                   0.2.9
Babel                         2.10.2
beautifulsoup4                4.11.1
Brlapi                        0.8.3
build                         0.8.0
CacheControl                  0.12.11
charset-normalizer            2.1.0
codespell                     2.2.1
cssselect                     1.1.0
cssutils                      2.6.0
cycler                        0.11.0
Cython                        0.29.32
dict2css                      0.3.0
distro                        1.7.0
docutils                      0.18.1
domdf-python-tools            3.3.0
enum-tools                    0.9.0.post1
extras                        1.0.0
fixtures                      4.0.0
fonttools                     4.37.1
gpg                           1.17.1-unknown
html5lib                      1.1
idna                          3.4
imagesize                     1.4.1
importlib-metadata            4.12.0
iniconfig                     1.1.1
Jinja2                        3.1.1
kiwisolver                    1.4.4
libcomps                      0.1.19
linkify-it-py                 2.0.0
lockfile                      0.12.2
louis                         3.23.0
lxml                          4.9.1
markdown-it-py                2.1.0
MarkupSafe                    2.1.1
matplotlib                    3.5.3
mdit-py-plugins               0.3.0
mdurl                         0.1.1
msgpack                       1.0.4
myst-parser                   0.18.0
natsort                       8.0.2
numpy                         1.23.1
olefile                       0.46
packaging                     21.3
pbr                           5.9.0
pep517                        0.12.0
Pillow                        9.2.0
pip                           22.2.2
platformdirs                  2.5.2
pluggy                        1.0.0
py                            1.11.0
Pygments                      2.13.0
PyGObject                     3.42.2
pyparsing                     3.0.9
pytest                        7.1.3
python-dateutil               2.8.2
pytz                          2022.1
PyYAML                        6.0
requests                      2.28.1
rpm                           4.17.0
ruamel.yaml                   0.17.21
ruamel.yaml.clib              0.2.6
scour                         0.38.2
setuptools                    65.3.0
six                           1.16.0
snowballstemmer               2.2.0
soupsieve                     2.3.2.post1
Sphinx                        5.1.1
sphinx-autodoc-typehints      1.19.2
sphinx-jinja2-compat          0.2.0
sphinx-prompt                 1.4.0
sphinx-rtd-theme              1.0.0
sphinx-tabs                   3.4.1
sphinx-toolbox                3.2.0
sphinxcontrib-applehelp       1.0.2.dev20220730
sphinxcontrib-devhelp         1.0.2.dev20220730
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1.dev20220730
sphinxcontrib-qthelp          1.0.3.dev20220730
sphinxcontrib-serializinghtml 1.1.5
tabulate                      0.8.10
testtools                     2.5.0
tomli                         2.0.1
tornado                       6.2
typing_extensions             4.3.0
uc-micro-py                   1.0.1
urllib3                       1.26.12
webencodings                  0.5.1
wheel                         0.37.1
zipp                          3.8.1
```
</details>

After add those two units to --deselect list another unit popped out.
<details>

```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra --import-mode=importlib --deselect 'zmq/tests/test_cython.py::test_cython[1]' --deselect 'zmq/tests/test_cython.py::test_cython[2]'
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.14, pytest-7.1.3, pluggy-1.0.0
rootdir: /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0, configfile: pytest.ini, testpaths: zmq/tests
collected 357 items / 1 deselected / 1 skipped / 356 selected

zmq/tests/test_asyncio.py ....................................                                                                                                       [ 10%]
zmq/tests/test_auth.py .................                                                                                                                             [ 14%]
zmq/tests/test_cffi_backend.py ssssssssssssss                                                                                                                        [ 18%]
zmq/tests/test_constants.py ..........................................                                                                                               [ 30%]
zmq/tests/test_context.py ....................s......                                                                                                                [ 38%]
zmq/tests/test_cython.py F                                                                                                                                           [ 38%]
zmq/tests/test_decorators.py ...........................                                                                                                             [ 46%]
zmq/tests/test_device.py ........                                                                                                                                    [ 48%]
zmq/tests/test_draft.py ss                                                                                                                                           [ 48%]
zmq/tests/test_error.py ...                                                                                                                                          [ 49%]
zmq/tests/test_etc.py .ss                                                                                                                                            [ 50%]
zmq/tests/test_ext.py .                                                                                                                                              [ 50%]
zmq/tests/test_future.py ....................                                                                                                                        [ 56%]
zmq/tests/test_imports.py ..........s.s.s.s.s.s.s                                                                                                                    [ 62%]
zmq/tests/test_includes.py ..                                                                                                                                        [ 63%]
zmq/tests/test_ioloop.py ....                                                                                                                                        [ 64%]
zmq/tests/test_log.py ........                                                                                                                                       [ 66%]
zmq/tests/test_message.py ....................                                                                                                                       [ 72%]
zmq/tests/test_monitor.py ......                                                                                                                                     [ 74%]
zmq/tests/test_monqueue.py ........                                                                                                                                  [ 76%]
zmq/tests/test_multipart.py ..                                                                                                                                       [ 76%]
zmq/tests/test_pair.py ....                                                                                                                                          [ 78%]
zmq/tests/test_poll.py ........                                                                                                                                      [ 80%]
zmq/tests/test_proxy_steerable.py ...                                                                                                                                [ 81%]
zmq/tests/test_pubsub.py ..                                                                                                                                          [ 81%]
zmq/tests/test_reqrep.py ......                                                                                                                                      [ 83%]
zmq/tests/test_retry_eintr.py s...s.                                                                                                                                 [ 85%]
zmq/tests/test_security.py .....                                                                                                                                     [ 86%]
zmq/tests/test_socket.py .............................s.....                                                                                                         [ 96%]
zmq/tests/test_ssh.py .                                                                                                                                              [ 96%]
zmq/tests/test_version.py ....                                                                                                                                       [ 97%]
zmq/tests/test_win32_shim.py s                                                                                                                                       [ 98%]
zmq/tests/test_z85.py ....                                                                                                                                           [ 99%]
zmq/tests/test_zmqstream.py ...                                                                                                                                      [100%]

================================================================================= FAILURES =================================================================================
______________________________________________________________________________ test_cython[3] ______________________________________________________________________________

language_level = 3, request = <FixtureRequest for <Function test_cython[3]>>, tmpdir = local('/tmp/pytest-of-tkloczko/pytest-998/test_cython_3_0')

    @pytest.mark.skipif(
        'zmq.backend.cython' not in sys.modules, reason="Requires cython backend"
    )
    @pytest.mark.skipif(
        sys.platform.startswith('win'), reason="Don't try runtime Cython on Windows"
    )
    @pytest.mark.parametrize('language_level', [3, 2])
    def test_cython(language_level, request, tmpdir):
        import pyximport

        assert 'zmq.tests.cython_ext' not in sys.modules

        importers = pyximport.install(
            setup_args=dict(include_dirs=zmq.get_includes()),
            language_level=language_level,
            build_dir=str(tmpdir),
        )

        cython_ext = None

        def unimport():
            pyximport.uninstall(*importers)
            sys.modules.pop('zmq.tests.cython_ext', None)

        request.addfinalizer(unimport)

        # this import tests the compilation
>       from . import cython_ext
E       ImportError: cannot import name 'cython_ext' from 'zmq.tests' (/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/zmq/tests/__init__.py)

zmq/tests/test_cython.py:35: ImportError
============================================================================= warnings summary =============================================================================
zmq/tests/test_poll.py:160
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_poll.py:160: PytestUnknownMarkWarning: Unknown pytest.mark.flaky - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @mark.flaky(reruns=3)

zmq/tests/test_poll.py:195
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_poll.py:195: PytestUnknownMarkWarning: Unknown pytest.mark.flaky - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @mark.flaky(reruns=3)

zmq/tests/test_retry_eintr.py:60
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_retry_eintr.py:60: PytestUnknownMarkWarning: Unknown pytest.mark.flaky - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @mark.flaky(reruns=3)

zmq/tests/test_auth.py: 10 warnings
zmq/tests/test_future.py: 20 warnings
zmq/tests/test_ioloop.py: 1 warning
zmq/tests/test_zmqstream.py: 3 warnings
  /usr/lib64/python3.8/site-packages/tornado/ioloop.py:350: DeprecationWarning: make_current is deprecated; start the event loop first
    self.make_current()

zmq/tests/test_context.py::TestContext::test_fail_init
  /usr/lib/python3.8/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function Context.__del__ at 0x7fb4d67479d0>

  Traceback (most recent call last):
    File "/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/zmq/sugar/context.py", line 93, in __del__
      self.destroy()
    File "/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/zmq/sugar/context.py", line 258, in destroy
      sockets = self._sockets
    File "/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/zmq/sugar/attrsettr.py", line 48, in __getattr__
      raise AttributeError(
  AttributeError: Context has no such option: _SOCKETS

    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))

zmq/tests/test_cython.py::test_cython[3]
  /usr/lib64/python3.8/site-packages/pyximport/pyximport.py:51: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

zmq/tests/test_future.py: 20 warnings
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_future.py:25: DeprecationWarning: make_current is deprecated; start the event loop first
    self.loop.make_current()

zmq/tests/test_future.py: 20 warnings
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_future.py:32: DeprecationWarning: clear_current is deprecated
    IOLoop.clear_current()

zmq/tests/test_future.py: 20 warnings
zmq/tests/test_ioloop.py: 4 warnings
  /usr/lib64/python3.8/site-packages/tornado/ioloop.py:227: DeprecationWarning: clear_current is deprecated
    IOLoop.clear_current()

zmq/tests/test_ioloop.py::TestIOLoop::test_close_all
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_ioloop.py:111: DeprecationWarning: zmq.eventloop.ioloop is deprecated in pyzmq 17. pyzmq now works with default tornado and asyncio eventloops.
    loop = self.IOLoop.current()

zmq/tests/test_ioloop.py::TestIOLoop::test_close_all
zmq/tests/test_ioloop.py::TestIOLoop::test_current
zmq/tests/test_ioloop.py::TestIOLoop::test_instance
zmq/tests/test_ioloop.py::TestIOLoop::test_simple
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_ioloop.py:74: DeprecationWarning: clear_current is deprecated
    BaseIOLoop.clear_current()

zmq/tests/test_ioloop.py::TestIOLoop::test_simple
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_ioloop.py:80: DeprecationWarning: make_current is deprecated; start the event loop first
    loop.make_current()

zmq/tests/test_monitor.py::TestSocketMonitorAsyncIO::test_monitor
zmq/tests/test_monitor.py::TestSocketMonitorAsyncIO::test_monitor_connected
  /home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/zmq/_future.py:663: RuntimeWarning: No running event loop. zmq.asyncio should be used from within an asyncio loop.
    self._update_handler(self._state)

zmq/tests/test_socket.py::TestSocket::test_int_sockopts
zmq/tests/test_socket.py::TestSocket::test_int_sockopts
zmq/tests/test_socket.py::TestSocket::test_mockable
  /home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/zmq/sugar/socket.py:261: DeprecationWarning: Socket.socket_type is deprecated, use Socket.type
    warn("Socket.socket_type is deprecated, use Socket.type", DeprecationWarning)

zmq/tests/test_zmqstream.py::TestZMQStream::test_callable_check
zmq/tests/test_zmqstream.py::TestZMQStream::test_on_recv_basic
zmq/tests/test_zmqstream.py::TestZMQStream::test_on_recv_wake
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_zmqstream.py:29: DeprecationWarning: make_current is deprecated; start the event loop first
    self.loop.make_current()

zmq/tests/test_zmqstream.py::TestZMQStream::test_callable_check
zmq/tests/test_zmqstream.py::TestZMQStream::test_on_recv_basic
zmq/tests/test_zmqstream.py::TestZMQStream::test_on_recv_wake
  /home/tkloczko/rpmbuild/BUILD/pyzmq-24.0.0/zmq/tests/test_zmqstream.py:39: DeprecationWarning: clear_current is deprecated
    ioloop.IOLoop.clear_current()

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================= short test summary info ==========================================================================
SKIPPED [1] zmq/tests/test_mypy.py:12: could not import 'mypy': No module named 'mypy'
SKIPPED [1] zmq/tests/test_cffi_backend.py:83: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:93: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:34: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:62: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:133: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:108: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:122: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:115: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:210: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:174: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:146: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:49: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:40: CFFI not available
SKIPPED [1] zmq/tests/test_cffi_backend.py:29: CFFI not available
SKIPPED [1] zmq/tests/test_context.py:351: Requires pyczmq
SKIPPED [2] zmq/tests/test_draft.py:15: draft api unavailable
SKIPPED [1] zmq/tests/test_etc.py:17: bundled libzmq
SKIPPED [1] zmq/tests/test_etc.py:23: bundled libzmq
SKIPPED [7] zmq/tests/test_imports.py:93: could not import 'zmq.green': No module named 'gevent'
SKIPPED [1] zmq/tests/test_retry_eintr.py:90: TODO: find a way to interrupt getsockopt
SKIPPED [1] zmq/tests/test_retry_eintr.py:93: TODO: find a way to interrupt setsockopt
SKIPPED [1] zmq/tests/test_socket.py:539: Requires pyczmq
SKIPPED [1] zmq/tests/test_win32_shim.py:25: Windows only test
FAILED zmq/tests/test_cython.py::test_cython[3] - ImportError: cannot import name 'cython_ext' from 'zmq.tests' (/home/tkloczko/rpmbuild/BUILDROOT/python-zmq-24.0.0-2.fc...
============================================= 1 failed, 325 passed, 31 skipped, 1 deselected, 120 warnings in 67.85s (0:01:07) =============================================
```
</details>

Looks like there are some dependencies between units. This kind of issue could be exposed usnig `pytest-randomly` module https://github.com/mrbean-bremen/pytest-find-dependencies/

--- a/setup.py
+++ b/setup.py
@@ -1299,6 +1299,7 @@ package_data = {
     'zmq.backend.cffi': ['*.h', '*.c'],
     'zmq.devices': ['*.pxd'],
     'zmq.sugar': ['*.pyi'],
+    'zmq.tests': ['*.pyx'],
     'zmq.utils': ['*.pxd', '*.h', '*.json'],
 }
 

--- a/zmq/tests/test_cython.py
+++ b/zmq/tests/test_cython.py
@@ -1,10 +1,18 @@
+import os
 import sys
 
 import pytest
 
 import zmq
 
+HERE = os.path.dirname(__file__)
+cython_ext = os.path.join(HERE, "cython_ext.pyx")
 
+
+@pytest.mark.skipif(
+    not os.path.exists(cython_ext),
+    reason=f"Requires cython test file {cython_ext}",
+)
 @pytest.mark.skipif(
     'zmq.backend.cython' not in sys.modules, reason="Requires cython backend"
 )

